preamble.tex

```diff
@@ -26,7 +20,9 @@
 \usepackage{titling}
 \usepackage{verbatim}
 \usepackage{xspace}
-
+\PassOptionsToPackage{hyphens}{url}
+\usepackage{hyperref} % allow for URL wrapping
+\usepackage{cleveref}
```

Comment:
Please keep packages in alphabetical order

Reply:
cleveref is an extension to hyperref it needs  be loaded afterwards 
In fact, I tried my best to stick to the documentation of each package here. 
In general hyperref is to be loaded last except for a few packages like cleveref. 
FYI hyperref redefines circa 50% of LaTeX, hence this strange loading order